File: minimumDifference.scala
import java.util.Arrays
object Solution {
  def minimumDifference(nums: Array[Int], k: Int) = {
    Arrays.sort(nums)
    var ans = Int.MaxValue
    var i = 0
    while ( {
      i + k - 1 < nums.length
    }) {
      ans = Math.min(ans, nums(i + k - 1) - nums(i))
      i += 1
    }
    ans
  }
}

File: minimumDifference.c
int cmp(const void *a, const void *b){
    return *(int*)b - *(int*)a;
}
int minimumDifference(int* nums, int numsSize, int k){
    qsort(nums,numsSize,sizeof(int),cmp);
    int l = 0, r = k-1;
    int min = 100001;
    while(r < numsSize){
        min = nums[l] - nums[r] < min ? (nums[l] - nums[r]) : min;
        l++;
        r++;
    }
    return min;
}

